books/kestrel/abnf/examples/pdf-grammar.txt

```diff
@@ -0,0 +1,127 @@
+pdf = header objects xref trailer
+
+header = pdf-marker int period int ws
+
+pdf-marker = %x25.50.44.46.45 ; %PDF-
+
+objects =*object
+
+object = int ws zero obj-marker ws dictionary ws [ stream ] endobj-marker
+
+obj-marker = %x6f.62.6a ; obj
+
+endobj-marker = %x65.6e.64.6f.62.6a ; endobj
+
+stream = stream-marker *char endstream-marker
+
+stream-marker = %x73.74.72.65.61.6d ; stream
+
+endstream-marker = %x65.6e.64.73.74.72.65.61.6d ; endstream
+
+xref = xref-marker ws zero ws int ws *xref-entry ;
+
+xref-entry = xref-int ws xref-int ws use-marker ;
+
+use-marker = %x6e / %x66 ; n or f
+
+xref-marker = %x78.72.65.66 ; xref
+
+xref-int = *digit
+
+trailer = trailer-marker ws dictionary ws startxref-marker ws int eof-marker;
+
+startxref-marker = %x73.74.61.72.74.78.72.65.66 ; startxref
+
+eof-marker = %x25.25.45.4f.46 ; %%EOF
+
+trailer-dictionary = start-dict *trailer-key-and-value end-dict
+
+trailer-key-and-value = (%s"/Size" int) / (%s"/Root" reference) / (%s"/Info" reference) / (%s"/ID" left-bracket string string right-bracket)
+
+trailer-marker = %x74.72.61.69.6c.65.72 ;
+
+dictionary = start-dict *dict-entry end-dict ;
+
+dict-entry =  type-entry / reference-entry / reference-array-entry / rectange-entry / number-entry / name-entry / font-entry / default-entry 
+
+default-entry = dict-key ws dict-value ;
+
+type-entry = %s"/Type" name
+
+reference-entry = %s"/Parent" / %s"/Pages" / %s"/Contents" / %s"/Resources" reference
+
+reference-array-entry = %s"/Kids" reference-array
+
+rectange-entry = %s"/MediaBox" / %s"CropBox" rectangle 
+
+number-entry = (%s"/Count") / (%s"/Length") number
+
+name-entry = (%s"/Subtype") / (%s"/BaseFont") / (%s"/Encoding") name
+
+font-entry %s"/Font" font
+
+font = start-dict name reference end-dict
+
+dict-key = backslash *char ;
+
+dict-value = boolean / number / name / array / dictionary / null / rectange / reference / string ;
+
+start-dict = %x3c.3c        ; <<<
+
+end-dict = %x3e.3e          ; >>
+
+ws = *(
+	
```

Comment:
There's an extra blank line and the `%x00` is further right than the other codes. Can the layout be made more uniform?